### PR TITLE
Improved ci

### PR DESCRIPTION
**Pull request by AdrieanKhisbe**
- Adapt test to `ert-runner` convention
- Update travis to use containers and evm
- Setup coverage with undercover

:smiley: 

cf https://github.com/chrisbarrett/skeletor.el/pull/29#issuecomment-164114674

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,21 +1,20 @@
-language: emacs-lisp
-
+language: generic
+sudo: false
+before_install:
+  - curl -fsSkL https://gist.github.com/rejeep/ebcd57c3af83b049833b/raw > x.sh && source ./x.sh
```
Comment (bbatsov) [inline .travis.yml]: x.sh?

Comment (AdrieanKhisbe) [inline .travis.yml]: I can change it if you want :)

**Comment by AdrieanKhisbe** [comment]
Yep, those where made by @rejeep, targeting travis.

**Comment by AdrieanKhisbe** [comment]
> $touch PR #  =)

**Comment by taku0** [comment]
Some of this PR is Incorporated into https://github.com/swift-emacs/swift-mode/commit/78cea9d79c8055d5ee0032d9f3457d257dda96ee.

I could not setup undercover correctly. I inserted the following code before `(require 'swift-mode)` in `swift-mode-test-indent.el` but it did not produce `/tmp/local-report.json`.

```elisp
(require 'undercover)
(undercover "*.el" "../*.el" (:report-file "/tmp/local-report.json") (:send-report nil))
```

I believe the undercover is installed because `(require 'undercover)` does not raise error.

**Comment by AdrieanKhisbe** [comment]
Closing due to conflict and partial incorporation

**Pull request closed without merging**